src/proxy/socks5.rs

```diff
@@ -229,3 +249,53 @@ async fn handle(
     });
     Ok(())
 }
+
+async fn dns_lookup(
```

Comment:
Feels like this (or some of it) eventually belongs on the `resolver` itself, and not in `socks5`, but fine for now I guess.